pkg/provider/provider.go

```diff
@@ -310,3 +296,24 @@ func (p *externalMetricsProvider) selectGroupResource(namespace string) apischem
 func (p *externalMetricsProvider) getMetricNameWithNamespace(metricName string) string {
 	return strings.Join([]string{p.namespace, metricName}, "|")
 }
+
+// TODO: remove this function once cli move it from internal module to pkg
+func parseScope(entity *swctlapi.Entity) swctlapi.Scope {
```

Comment:
@kezhenxu94 this func is copied from cli repo. Could we move it from internal to pkg?

Reply:
> @kezhenxu94 this func is copied from cli repo. Could we move it from internal to pkg?

@fgksgf can you take over this? Thanks

Reply:
@kezhenxu94 No problem, i will do this today